extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -1105,8 +1106,14 @@ export class AmpStoryPage extends AMP.BaseElement {
   emitProgress_(progress) {
     // Don't emit progress for ads, since the progress bar is hidden.
     // Don't emit progress for inactive pages, because race conditions.
+    const storyAdSegmentBranch = getExperimentBranch(
+      this.win,
+      StoryAdSegmentExp.ID
+    );
     if (
-      (!isExperimentOn(this.win, 'story-ad-auto-advance') && this.isAd()) ||
+      ((!storyAdSegmentBranch ||
```

Comment:
nit: these complex if statements here and below are a bit hard to parse, maybe pulling to a named var will help.